### PR TITLE
Simplify Service Client Configuration Resolving

### PR DESCRIPTION
**Pull request by skmcgrail**
Further simplifies the resolution of service specific client configuration from external sources.

### CODE REVIEW

File: aws/external/config_test.go [hunk 1]
```diff
@@ -101,4 +103,13 @@ func TestConfigs_ResolveAWSConfig(t *testing.T) {
...
 		t.Errorf("expect %v provider, got %v", e, a)
 	}
+
+	var expectedSources []interface{}
+	for _, s := range cfg.ConfigSources {
+		expectedSources = append(expectedSources, s)
+	}
+
+	if e, a := expectedSources, cfg.ConfigSources; !reflect.DeepEqual(e, a) {
+		t.Errorf("expected %v, got %v", e, a)
```
Comment (skotambkar) [inline aws/external/config_test.go]: QQ: When to use `reflect.DeepEqual` vs `cmp.Diff`?

I thought cmp.Diff was preferred, but for []interface not sure

File: private/model/api/api.go [hunk 2]
```diff
@@ -699,10 +699,8 @@ func New(config aws.Config) *{{ .StructName }} {
     }
 
 	{{ if .HasExternalClientConfigFields }}
-		if config.AdditionalConfig != nil {
-			if err := config.AdditionalConfig.ResolveConfig(resolveClientConfig(svc)); err != nil {
-				panic(fmt.Errorf("failed to resolve service configuration: %v", err))
-			}
+		if err := resolveClientConfig(svc, config.ConfigSources); err != nil {
+			panic(fmt.Errorf("failed to resolve service configuration: %v", err))
```
Comment (skotambkar) [inline private/model/api/api.go]: Should we use log.Fatalf here instead of panic?

Comment (skmcgrail) [reply private/model/api/api.go]: This solution is only temporary until we implement the client builder pattern. Today the client.New(...) operation does not have the ability to return an error. So we opt for panic in this case.

File: aws/external/config_test.go (hunk 1 above)
Comment (skmcgrail) [reply aws/external/config_test.go]: I think this is a much bigger discussion we can take offline, but in essence cmp.Diff has much more granular comparisons, for example know that two timestamps are equal across timezones. Additionally it has the ability to set options like ignoring/not-ignoring unexported structure fields. In this instance of comparison the need for using cmp.Diff is unwarranted.

**Pull request merged**